src/pytorch_lightning/trainer/connectors/callback_connector.py

```diff
@@ -290,3 +294,18 @@ def _configure_external_callbacks() -> List[Callback]:
         )
         external_callbacks.extend(callbacks_list)
     return external_callbacks
+
+
+def _validate_callbacks_list(callbacks: List[Callback]) -> None:
+    stateful_callbacks = [cb for cb in callbacks if is_overridden("state_dict", instance=cb, parent=Callback)]
```

Comment:
```suggestion
    stateful_callbacks = [cb for cb in callbacks if is_overridden("state_dict", cb)]
```

Reply:
I think we need the parent here. What if we have this inheritance: A > B > C where A is the super class of B and B is super class of C. If B implements state_dict, but C doesn't, I still want to categorize this as stateful.

Reply:
Oh, you are thinking that `is_overridden` is smarter than it is and actually checks the inheritance tree. Not the case: https://github.com/Lightning-AI/lightning/blob/43fd9c1534411a80d5364031b3e529fef02a930c/src/pytorch_lightning/utilities/model_helpers.py#L31-L32

Reply:
Double oh, I was importing from lightning_utilities and this one isn't smart. Agree, I should be using the one from PL and then I can drop the parent! Great!